Modules/SubModule_Import/xASL_imp_NII2BIDS_Run.m

```diff
@@ -1,59 +1,89 @@
-% DESCRIPTION: NII2BIDS conversion for a single sessions, single runs.
-%
-% 1. Convert anat files
-% 2. Convert perf files
+% DESCRIPTION: NII2BIDS conversion for a single run.
+% 
+% 1. Make a subject directory with a correct session name
+% 2. Convert structural and ASL runs
 %
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
-% EXAMPLE:     x = xASL_imp_NII2BIDS_Run(x, bidsPar, studyPar, subjectSessionLabel, inSessionPath, outSessionPath, listRuns, iRun);
+% EXAMPLE:     x = xASL_imp_NII2BIDS_Run(x, imPar, bidsPar, studyPar, listRuns, nameSubjectSession, bidsLabel, iRun);
 % __________________________________
 % Copyright 2015-2021 ExploreASL
 
+    %% Print the run that is being converted
+    fprintf('\n[\b======================================== CONVERT RUN =========================================]\b\n');
+    fprintf('Converting subject %s, session %s, run %s ', bidsLabel.subject, bidsLabel.visit, listRuns{iRun});
 
-    %% Anatomical files
-    fprintf('run %d ...\n', iRun);
-    try
-        xASL_imp_NII2BIDS_RunAnat(imPar, bidsPar, studyPar, subjectSessionLabel, outSessionPath, listRuns, iRun, nameSubjectSession);
-    catch loggingEntry
-        [x] = xASL_qc_AddLoggingInfo(x, loggingEntry);
-        xASL_imp_NII2BIDS_RunIssueWarning(loggingEntry, 'anatomical', subjectSessionLabel, iRun);
-    end
-    
-    %% Perfusion files
-    try
-        xASL_imp_NII2BIDS_RunPerf(imPar, bidsPar, studyPar, subjectSessionLabel, inSessionPath, outSessionPath, listRuns, iRun);
-    catch loggingEntry
-        [x] = xASL_qc_AddLoggingInfo(x, loggingEntry);
-        xASL_imp_NII2BIDS_RunIssueWarning(loggingEntry, 'perfusion', subjectSessionLabel, iRun);
+    %% 1. Make a subject directory with a correct session name
+    if ~isempty(bidsLabel.visit)
+        %if ~isempty(bidsLabel.visit)
+        sessionLabel = ['ses-' bidsLabel.visit];
+        %else
+        %    sessionLabel = ['ses-' listRuns{iRun}(5:end)];
+        %end
+        
+        % Create directory if it does not exist already
+        sessionPerfusionDirectory = fullfile(imPar.BidsRoot,['sub-' bidsLabel.subject], sessionLabel,'perf');
+        xASL_adm_CreateDir(sessionPerfusionDirectory);
+        
+        inSessionPath = fullfile(imPar.TempRoot, nameSubjectSession, listRuns{iRun});
+        outSessionPath = fullfile(imPar.BidsRoot, ['sub-' bidsLabel.subject], sessionLabel);
+
+        % Need to add the underscore so that it doesn't need to be added automatically and can be skipped for empty session
+        sessionLabel = ['_' sessionLabel];
+    else
+        % Session label is skipped
+        sessionLabel = '';
+
+        % Only one session - no session labeling
+        if ~exist(fullfile(imPar.BidsRoot,['sub-' bidsLabel.subject]),'dir')
```

Comment:
Existing-checking can be removed, not required for `xASL_adm_CreateDir`

Reply:
Fixed.